pysys/baserunner.py

```diff
@@ -160,6 +164,17 @@ def __init__(self, record, purge, cycle, mode, threads, outsubdir, descriptors,
 		else:
 			self.writers.append(ConsoleSummaryResultsWriter())
 		
+		self.__collectTestOutput = []
+		for c in self.project.collectTestOutput:
+			c = dict(c)
+			assert c['outputDir'], 'collect-test-output outputDir cannot be empty'
```

Comment:
Do you also need to check that pattern and outputPattern exist here?

Reply:
no I think not - I put that assert in not as part of checking that the config is sane, but just as an emergency failsafe to prevent deleting the current directory (!) which is what accidentally happened at one point - so it's really to protect to following deletedir line (locality of reference is a helpful property for assertions). 

Detailed checking that the pattern and other attributes are set happens in project.py line 237 and following. The assert you spotted should never be needed, but given the seriousness of deleting "." I felt it was worth retaining.